packages/yoroi-extension/chrome/extension/ergo-connector/utils.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import type { Tx } from './types';
+import type { TokenRow } from '../../../app/api/ada/lib/storage/database/primitives/tables';
+import { Logger } from '../../../app/utils/logging';
+
+function parseEIP0004Data(input: any): ?string {
```

Comment:
I didn't use the vlq npm package here since:
1) it uses base64url
2) we need to partially read the string, not the entire thing, and the use the intermediate info of how far we parsed into the vlq-encoded length to then read the remainder of the string